components/FutureGraph.tsx

```diff
@@ -0,0 +1,318 @@
+import {
+  ResponsiveContainer,
+  YAxis,
+  AreaChart,
+  LabelList,
+  Area,
+  ReferenceLine,
+} from 'recharts';
+import useSWR, { Fetcher } from 'swr';
+import {
+  FutureThresholdLegendIcon,
+  UpArrowIcon,
+  DownArrowIcon,
+  LineCircle,
+} from './icons/future_threshold';
+interface ForecastDataPointProps {
+  target_datetime_utc: number;
+  expected_generation_kw: number;
+}
+
+interface ForecastDataProps {
+  forecast_uuid: string;
+  site_uuid: string;
+  forecast_creation_datetime: number;
+  forecast_version: string;
+  forecast_values: ForecastDataPointProps[];
+}
+
+interface UnparsedForecastDataPointProps {
+  target_datetime_utc: string | number;
+  expected_generation_kw: number;
+}
+
+interface UnparsedForecastDataProps {
+  forecast_uuid: string;
+  site_uuid: string;
+  forecast_creation_datetime: string | number;
+  forecast_version: string;
+  forecast_values: UnparsedForecastDataPointProps[];
+}
+
+const formatter = new Intl.DateTimeFormat(['en-US', 'en-GB'], {
+  hour: 'numeric',
+  minute: 'numeric',
+});
+
+const fetcher: Fetcher<ForecastDataProps> = async (url: string) => {
+  const tempData: UnparsedForecastDataProps = await fetch(url).then((res) =>
+    res.json()
+  );
+
+  if (typeof tempData.forecast_creation_datetime === 'string') {
+    tempData.forecast_creation_datetime = Date.parse(
+      tempData.forecast_creation_datetime
+    );
+  } else {
+    throw new Error('Data contains values with incompatible types');
+  }
+
+  tempData.forecast_values.map(({ target_datetime_utc }) => {
+    if (typeof target_datetime_utc === 'string') {
+      target_datetime_utc = Date.parse(target_datetime_utc);
+    } else {
+      throw new Error('Data contains values with incompatible types');
+    }
+  });
+  return tempData as ForecastDataProps;
+};
+
+const siteUUID = 'b97f68cd-50e0-49bb-a850-108d4a9f7b7e';
+
+/* Represents the threshold (2000 kWh) for the graph */
+const graphThreshold = 0.7;
+
+const FutureGraph = () => {
+  const { data, isLoading } = useSWR(
+    `${process.env.NEXT_PUBLIC_API_BASE_URL}/api/sites/pv_forecast/${siteUUID}`,
+    fetcher
+  );
+
+  /* The index of the current time in the forecast data */
+  const getCurrentTimeForecastIndex = () => {
+    return 9;
+  };
+
+  /**
+   * Renders two types of labels for the graph:
+   * - A circle for the current time
+   * - A threshold label
+   * @param props data about the point, such as x and y position on the graph
+   * @returns SVG element
+   */
+  const renderGraphLabels = (props: any) => {
```

Comment:
I think this function could be split into 2 separate ones since it has 2 distinct goals. Maybe this current function could just show the line circle label, and the label for the threshold could be put into another function, which is then passed to another `<LabelList>` component. If that works, that is, I haven't tested.